japicmp/src/main/java/japicmp/cmp/JarArchiveComparator.java

```diff
@@ -296,7 +296,6 @@ public List<CtClass> filterClasses(List<CtClass> ctClasses, ReducibleClassPool c
 						LOGGER.fine(String.format("Adding class '%s' with jar name '%s' to list.", ctClass.getName(), ctClass.getName()));
 					}
 				} else {
-					classPool.remove(ctClass);
```

Comment:
I don't think this is necessary. The issue is about old code that no longer is called / has been removed.

Reply:
Using 0.18.5 we get error: Execution default-cli of goal com.github.siom79.japicmp:japicmp-maven-plugin:0.18.5:cmp failed: Could not load 'com.***.***.***.***.C***': com.***.***.***.***.C***. Please make sure that all libraries have been added to the classpath (CLASSPATH=:/r***...:) or try the option '--ignore-missing-classes'.
After removing above line it starts to work again. Please check.

Reply:
Do you have a concrete example to reproduce? The message `Please make sure that all libraries have been added to the classpath ...` is normally printed when the class is NOT on the classpath. It would be an error if it actually is on the classpath; hence, I would need a concrete example.

Reply:
Btw: The code here filters the classes by the configured filters. Are you sure that there is not filter configured, that excludes the mentioned class?